DSpace/DSpace
# Find all bugfix and media filter update

## Description
This PR adds a separate sorting option to the item find all methods to be used with the limits and offsets. 
The findAll methods that do not use an offset and limit have been renamed to "read only" to indicate that these methods should not be used for batch edits. 

The MediaFilterServiceImpl has been updated to use the findAll methods with offset and limit. 

## Instructions for Reviewers
Verify that the sorting works on the findAll methods. 
* Retrieve items in batches using the pagination in the rest API. 
* Verify that items are retrieved in order based on the sorting number assigned in the database

Test the filtermedia process
* Run the filtermedia process on all items in a repository
* Verify that the items are also retrieved and processed according to the sorting number
* Verify all items are processed


## Checklist
